File: app/src/main/java/com/foxek/simpletimer/ui/workout/dialog/WorkoutCreateDialog.java
package com.foxek.simpletimer.ui.workout.dialog;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;
import android.widget.TextView;

import com.foxek.simpletimer.R;
import com.foxek.simpletimer.di.component.ActivityComponent;
import com.foxek.simpletimer.ui.base.BaseFragment;
import com.foxek.simpletimer.ui.workout.WorkoutContact;

import javax.inject.Inject;

import androidx.annotation.NonNull;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import butterknife.Unbinder;

public class WorkoutCreateDialog extends BaseFragment {

    @Inject
    WorkoutContact.Presenter presenter;

    @BindView(R.id.error_hint)
    TextView errorHint;

    @BindView(R.id.training_edit_text)
    EditText workoutEditText;

    private Unbinder binder;

    public static WorkoutCreateDialog newInstance() {
        return new WorkoutCreateDialog();
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {

        View dialogView = inflater.inflate(R.layout.dialog_add_workout, container, false);

        ActivityComponent component = getActivityComponent();

        if (component != null) {
            component.inject(this);
            binder = ButterKnife.bind(this, dialogView);
        }

        return dialogView;
    }

    @OnClick(R.id.create_button)
    void onCreateButtonClick() {
        if (!workoutEditText.getText().toString().isEmpty()) {
            workoutEditText.setCursorVisible(false);
            dismiss();
            presenter.saveButtonClicked(workoutEditText.getText().toString());
        } else {
            errorHint.setVisibility(View.VISIBLE);
        }
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        workoutEditText.setCursorVisible(false);
        binder.unbind();
    }
}

File: app/src/main/java/com/foxek/simpletimer/ui/workout/dialog/WorkoutCreateDialog.kt
package com.foxek.simpletimer.ui.workout.dialog

import android.os.Bundle
import android.view.View

import com.foxek.simpletimer.R
import com.foxek.simpletimer.ui.base.BaseDialog
import com.foxek.simpletimer.ui.workout.WorkoutContact

import javax.inject.Inject

import kotlinx.android.synthetic.main.dialog_add_workout.*

class WorkoutCreateDialog : BaseDialog() {

    override val dialogTag = "WorkoutCreateDialog"
    override val layoutId = R.layout.dialog_add_workout

    @Inject
    lateinit var presenter: WorkoutContact.Presenter

    companion object {
        fun newInstance(): WorkoutCreateDialog = WorkoutCreateDialog()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        activityComponent?.inject(this)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        dialog_add_workout_create_btn.setOnClickListener { createButtonClick() }
    }

    private fun createButtonClick(){
        if (checkNotEmpty(dialog_add_workout_field_name)) {
            dismiss()
            presenter.saveButtonClicked(dialog_add_workout_field_name.text.toString())
        } else {
            dialog_add_workout_error_hint.visibility = View.VISIBLE
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        dialog_add_workout_field_name.isCursorVisible = false
    }
}
